src/amber/router/cookies.cr

```diff
@@ -11,15 +11,19 @@ module Amber::Router
       @options = {} of String => String
 
       def permanent
-        @permanent ||= PermanentStore.new(self, @key_generator, @options)
+        @permanent ||= PermanentStore.new(self, @options)
       end
 
       def encrypted
-        @encrypted ||= EncryptedStore.new(self, @key_generator, @options)
+        @encrypted ||= EncryptedStore.new(self, @options)
       end
 
       def signed
-        @signed ||= SignedStore.new(self, @key_generator, @options)
+        @signed ||= SignedStore.new(self, @options)
+      end
+
+      def secret
+        @secret ||= Amber::Server.settings.secret_key_base || SecureRandom.urlsafe_base64(32) 
```

Comment:
Can we pass in on initialize `Amber::Server.settings.secret_key_base` or a property something like:

```crystal
@secret = Amber::Server.settings.secret_key_base
# or 
def initialize(@host = nil, @secure = false, @secret = Amber::Server.settings.secret_key_base)
```

I would also consider removing `|| SecureRandom.urlsafe_base64(32)` since secret_key_base always exists. I think having this can cause the same effect you are trying to resolve with this PR.

Reply:
:+1: